### ISSUE TITLE
Parking mode

### ISSUE THREAD

**Issue by izvvv** [Farama-Foundation/HighwayEnv#572]
I want to achieve side parking. In addition to customizing a suitable parking environment, how should I control the vehicle to achieve side parking?

**Comment by izvvv** [comment]
![image](https://github.com/Farama-Foundation/HighwayEnv/assets/78134574/c44677ec-5b57-47f4-a355-a6e86cd1dc6b)
For example, in this case, I only modified the road style, but the vehicle was carried out according to the goal of reversing into the warehouse every time.

**Comment by eleurent** [comment]
You should edit the environment definition, and in particular the part where the goal is defined:

https://github.com/Farama-Foundation/HighwayEnv/blob/81c11d6ff50a4fecf05a8341edaecfc245924e0b/highway_env/envs/parking_env.py#L193

By default, parking spots are defined as small lane segments and we use their direction as a target heading (i.e. vertical).
If you want side parking, you need to change this `goal.heading` definition to be horizontal.

**Comment by izvvv** [comment]
![image](https://github.com/Farama-Foundation/HighwayEnv/assets/78134574/df89dec2-40dc-497b-8755-3fcd0af13b2d)

**Comment by izvvv** [comment]
请问为什么有两组线，上方这一组从哪里来的？
![image](https://github.com/Farama-Foundation/HighwayEnv/assets/78134574/d8cb9b05-40a3-4c5f-b880-94c89b4c2681)


> 您应该编辑环境定义，特别是定义目标的部分：
> 
> https://github.com/Farama-Foundation/HighwayEnv/blob/81c11d6ff50a4fecf05a8341edaecfc245924e0b/highway_env/envs/parking_env.py#L193
> 
> 默认情况下，停车位被定义为小车道段，我们使用它们的方向作为目标航向（即垂直）。 如果你想要侧边停车，你需要将此`goal.heading`定义更改为水平（例如添加或删除np.pi/2）

**Comment by izvvv** [comment]
> 您应该编辑环境定义，特别是定义目标的部分：
> 
> https://github.com/Farama-Foundation/HighwayEnv/blob/81c11d6ff50a4fecf05a8341edaecfc245924e0b/highway_env/envs/parking_env.py#L193
> 
> 默认情况下，停车被定义为小车道段，我们使用它们的方向目标航向（即水平）作为。 如果您想要侧边停车，您需要另外`goal.heading`定义更改为水平（例如添加或删除 np.pi /2 ）

May I ask why there are two sets of lines? Where does the top set come from?
![image](https://github.com/Farama-Foundation/HighwayEnv/assets/78134574/2e6a20a0-c12f-415c-b6d5-a879c160e367)

**Comment by izvvv** [comment]
> ![image](https://private-user-images.githubusercontent.com/78134574/307497523-c44677ec-5b57-47f4-a355-a6e86cd1dc6b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk0NzcyNDksIm5iZiI6MTcwOTQ3Njk0OSwicGF0aCI6Ii83ODEzNDU3NC8zMDc0OTc1MjMtYzQ0Njc3ZWMtNWI1Ny00N2Y0LWEzNTUtYTZlODZjZDFkYzZiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzAzVDE0NDIyOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA0Zjk4MjVhYTQ5NzY0ODc5OGY3Yjk1YTA3MzUxYWUzZGU1YjM5ZGJlMjliMjg1MTQwMjMxMTM5MjdhZTRjZGQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.hK-fE1v28wvWUfxhMXDGLJDXo4LRR9QsouZGMtz3DlM) For example, in this case, I only modified the road style, but the vehicle was carried out according to the goal of reversing into the warehouse every time.

Like this, a single parking space looks like a red circle, but I want to realize a parking space that looks like a green circle. How do I do it.
If I only make one side of adjacent parking spaces coincide with each other through calculation, there will always be a line that does not meet the requirements at the far left end.
![image](https://github.com/Farama-Foundation/HighwayEnv/assets/78134574/1283d6ee-32b9-40c4-b6d5-3a9a70d76fe4)

**Comment by eleurent** [comment]
> If I only make one side of adjacent parking spaces coincide with each other through calculation, there will always be a line that does not meet the requirements at the far left end.

Yes I would have suggested something like that. For the far left end, you could simply add a dummy final lane (also with a line on a single side) that cannot be used as a parking spot (a goal location) but is there just to display the last line marking?

Otherwise, you'll probably have to implement a custom lane type or line display code.

**Comment by izvvv** [comment]
![image](https://github.com/Farama-Foundation/HighwayEnv/assets/78134574/5120477f-e37e-4903-a7dd-a5dab6c6f6ce)
![image](https://github.com/Farama-Foundation/HighwayEnv/assets/78134574/d51da5bc-3bed-480a-aadd-cda3278723d2)
![image](https://github.com/Farama-Foundation/HighwayEnv/assets/78134574/0c3cd091-0dcc-4e1f-9101-191d6d7c58c8)
![image](https://github.com/Farama-Foundation/HighwayEnv/assets/78134574/63ad5cdf-e3f7-453b-bb80-c6ffd043d62d)

Why can't the control vehicle stop direction match the parking space?   
heading=goal_lane.heading+np.pi/3  I tried it too
